org.eclipse.lsp4e/src/org/eclipse/lsp4e/LoggingStreamConnectionProviderProxy.java

```diff
@@ -120,7 +120,7 @@ private String message(String direction, byte[] payload) {
 		builder.append(' ');
 		builder.append(id);
 		builder.append(":\n"); //$NON-NLS-1$
-		builder.append(payload);
+		builder.append(new String(payload));
```

Comment:
I think you should add explicit encoding (probably UTF-8\*) so that users who are using regional charset by default don't end up with corrupted messages, or worse unhandled exceptions thrown here.

Note that the call to getBytes() below has a similar issue.

\* Note that LSP support the encoding header and LSP4J support that too so if you are intercepting the raw stream that can cause issues if you don't also read that header. In practice I have no idea if anyone uses anything other than UTF-8, I hope they don't.